backend/entityservice/database/insertions.py

```diff
@@ -261,3 +259,28 @@ def get_created_runs_and_queue(db, project_id):
     if res is None:
         res = []
     return res
+
+
+def is_dataprovider_allowed_to_upload_and_lock(db, dp_id):
+    """
+    This method returns true if the dataprovider is allowed to upload her clks.
+    A data provider is allowed to upload clks if they have not yet been uploaded or if the upload is already in progress.
```

Comment:
I think you mean something different with the second if.
A DP is not allowed to upload if data has been uploaded, or if an upload is currently underway.